sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_lease.py

```diff
@@ -47,7 +48,7 @@ class DataLakeLeaseClient:  # pylint: disable=client-accepts-api-version-keyword
         This will be `None` if no lease has yet been acquired or modified."""
 
     def __init__(  # pylint: disable=missing-client-constructor-parameter-credential, missing-client-constructor-parameter-kwargs
-        self, client: Union["FileSystemClient", "DataLakeDirectoryClient", "DataLakeFileClient"],
+        self, client: Union["FileSystemClient", "PathClient"],
```

Comment:
What was the issue here? `PathClient` is not exported to users and therefore this isn't super helpful to them.

Reply:
The issue is`PathClient`'s `acquire_lease` API constructs `DataLakeLeaseClient` rather than `DataLakeDirectoryClient` or `DataLakeFileClient`. The only other client that constructs `DataLakeLeaseClient` is `FileSystemClient`.
![Screenshot 2024-10-29 at 11 47 27 PM](https://github.com/user-attachments/assets/7522bb2a-7cdb-4330-9ad1-4299f7d41d11)

I thought it might be more appropriate to use `PathClient` as a type hint, but I agree that since it's not exported, it might not be helpful.